src/dataaggregator/aggregator/aggregator.go

```diff
@@ -19,63 +19,67 @@ type Aggregator struct {
 	appChan            chan *appmetric.AppMonitor
 	policyPoller       *PolicyPoller
 	metricPollerCount  int64
-	metricPollerArray  []*MetricPoller
+	MetricPollerArray  []*MetricPoller
 	database           db.DB
 }
 
 func NewAggregator(logger lager.Logger, clock clock.Clock, policyPollerInterval time.Duration, database db.DB, metricCollectorUrl string, metricPollerCount int64) *Aggregator {
 	aggregator := &Aggregator{
-		logger:             logger,
+		logger:             logger.Session("Aggregator"),
 		metricCollectorUrl: metricCollectorUrl,
 		doneChan:           make(chan bool),
 		appChan:            make(chan *appmetric.AppMonitor, 10),
 		metricPollerCount:  metricPollerCount,
-		metricPollerArray:  []*MetricPoller{},
+		MetricPollerArray:  []*MetricPoller{},
 		database:           database,
 	}
-	aggregator.policyPoller = NewPolicyPoller(lager.NewLogger("policy-poller"), clock, policyPollerInterval, database, aggregator.ConsumeTrigger, aggregator.appChan)
+	aggregator.policyPoller = NewPolicyPoller(logger, clock, policyPollerInterval, database, aggregator.ConsumeTrigger, aggregator.appChan)
 
 	client := cfhttp.NewClient()
 	client.Transport.(*http.Transport).MaxIdleConnsPerHost = (int)(metricPollerCount)
 
 	var i int64
 	for i = 0; i < metricPollerCount; i++ {
-		pollerLogger := lager.NewLogger("metric-poller-" + strconv.FormatInt(i, 10))
-		poller := NewMetricPoller(metricCollectorUrl, pollerLogger, aggregator.appChan, aggregator.ConsumeAppMetric, client)
-		aggregator.metricPollerArray = append(aggregator.metricPollerArray, poller)
+		poller := NewMetricPoller(metricCollectorUrl, logger, aggregator.appChan, aggregator.ConsumeAppMetric, client)
+		aggregator.MetricPollerArray = append(aggregator.MetricPollerArray, poller)
 	}
 	return aggregator
 }
 func (a *Aggregator) ConsumeTrigger(triggerMap map[string]*policy.Trigger, appChan chan *appmetric.AppMonitor) {
+	if triggerMap == nil {
+		return
+	}
 	for appId, trigger := range triggerMap {
 		for _, rule := range trigger.TriggerRecord.ScalingRules {
 			appChan <- &appmetric.AppMonitor{
-				AppId:          appId,
-				MetricType:     rule.MetricType,
-				StatWindowSecs: rule.StatWindowSecs,
+				AppId:      appId,
+				MetricType: rule.MetricType,
+				StatWindow: rule.StatWindow,
 			}
 		}
 	}
 }
 func (a *Aggregator) ConsumeAppMetric(appMetric *appmetric.AppMetric) {
+	if appMetric == nil {
+		a.logger.Error("appmetric is nil", errors.New("Should not save a nil appmetric to database"))
```

Comment:
Do we expect nil or not? I am trying to understand what I would do if I received this error.
If it's just that we shouldn't save it, then don't log an error. If it is unexpected then log an error.